### PR TITLE
fix(react-datepicker-compat): Fix theming issues with all themes and WHCM

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR fixes the following issues:
* When clicking grid items, there were unwanted styles left behind if while clicking you move the mouse out of the control.
* Contrast issues with high contrast and dark theme.
* Not showing current selection when the date range is month.
* Contrast issues with marked days.
* Colors didn't feel cohesive and seemed like we had two themes in one.

## Before

image
image

## After

image
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/98cb97736517c9041c95e13bd7bb27d9c0c1daa4'>98cb97736517c9041c95e13bd7bb27d9c0c1daa4</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-datepicker-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.ts [hunk 1]
```diff
@@ -111,35 +108,34 @@ const useDayCellStyles = makeStyles({
...
+    },
+
     [`&:hover, &.${extraCalendarDayGridClassNames.hoverStyle}, &.${extraCalendarDayGridClassNames.pressedStyle}`]: {
       color: tokens.colorNeutralForeground1Static,
-      backgroundColor: tokens.colorBrandBackgroundInvertedSelected + ' !important',
+      backgroundColor: tokens.colorBrandBackgroundInvertedSelected,
       '@media (forced-colors: active)': {
         backgroundColor: 'Highlight!important',
         color: 'HighlightText!important',
       },
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.ts:132]: Are all these `!important` needed?

File: packages/react-components/react-datepicker-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.ts [hunk 2]
```diff
@@ -237,9 +242,16 @@ const useDayIsTodayStyles = makeStyles({
     '@media (forced-colors: active)': {
       backgroundColor: 'WindowText!important',
       ...shorthands.borderColor('WindowText!important'),
-      color: 'WindowText!important',
+      color: 'Window!important',
       forcedColorAdjust: 'none',
     },
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/CalendarDayGrid/useCalendarDayGridStyles.ts:247]: Ditto here.

File: packages/react-components/react-datepicker-compat/src/components/CalendarPicker/useCalendarPickerStyles.ts [hunk 3]
```diff
@@ -212,6 +217,11 @@ const useCurrentStyles = makeStyles({
     color: tokens.colorNeutralForegroundOnBrand + '!important',
     fontWeight: tokens.fontWeightSemibold,
 
+    '@media (forced-colors: active)': {
+      backgroundColor: 'WindowText',
+      color: 'Window!important',
+      forcedColorAdjust: 'none',
+    },
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/CalendarPicker/useCalendarPickerStyles.ts:224]: Ditto here.

**Review by khmakoto — APPROVED** [review]

**Pull request merged**